scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -226,62 +256,151 @@ private static void askInteractively(Console console, UserInfo userInfo, Action
 
-            storeGroups(groupFilePath, groupsMap);
         } catch (Throwable t) {
             exitWithErrorMessage("Unexpected error", null, t);
         }
     }
 
+    private static void deleteAccount(UserInfo userInfo, String loginFilePath, String groupFilePath, Properties props, Multimap<String, String> groupsMap) throws ManageUsersException {
+        if (userInfo.getLogin().isEmpty()) {
+            warnWithMessage("Provided username is empty, skipping.");
+            return;
+        }
+        if (!props.containsKey(userInfo.getLogin())) {
+            warnWithMessage("USER " + userInfo.getLogin() + " does not exist in login file : " + loginFilePath);
+        }
+        if (!groupsMap.containsKey(userInfo.getLogin())) {
+            warnWithMessage("USER " + userInfo.getLogin() + " does not exist in group file : " + groupFilePath);
+        }
+        props.remove(userInfo.getLogin());
+        groupsMap.removeAll(userInfo.getLogin());
+        System.out.println("Deleted user " + userInfo.getLogin());
+    }
+
+    private static void updateAccount(PublicKey pubKey, UserInfo userInfo, String loginFilePath, String groupFilePath, Properties props, Multimap<String, String> groupsMap, boolean bulkMode) throws ManageUsersException, KeyException {
+        if (userInfo.getLogin().isEmpty()) {
+            warnWithMessage("Provided username is empty, skipping.");
+            return;
+        }
+        if (!props.containsKey(userInfo.getLogin())) {
+            if (userInfo.getPassword() != null && userInfo.getGroups() != null) {
+                warnWithMessage("USER " + userInfo.getLogin() + " does not exist in login file : " + loginFilePath + ", create this user.");
+            } else {
+                if (bulkMode) {
+                    warnWithMessage("USER " + userInfo.getLogin() + " does not exist in login file : " + loginFilePath + " and not enough information were provided to create a new user, skipping.");
+                    return;
+                } else {
+                    exitWithErrorMessage("USER " + userInfo.getLogin() + " does not exist in login file : " + loginFilePath + " and not enough information were provided to create a new user.", null, null);
+                }
+            }
+        }
+        if (!Strings.isNullOrEmpty(userInfo.getPassword())) {
+            updateUserPassword(pubKey, userInfo.getLogin(), userInfo.getPassword(), props);
+            System.out.println("Changed password for user " + userInfo.getLogin());
+        }
+        if (userInfo.getGroups() != null) {
+            updateUserGroups(userInfo.getLogin(), userInfo.getGroups(), groupsMap);
+        }
+
+        System.out.println("Updated user " + userInfo.getLogin());
+    }
+
+    private static void createAccount(PublicKey pubKey, UserInfo userInfo, String loginFilePath, String groupFilePath, Properties props, Multimap<String, String> groupsMap, boolean bulkMode) throws ManageUsersException, KeyException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "bulkMode". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)